packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -113,36 +142,34 @@ export function getDefaultSpecFileName (specPattern: string, testingType: Testin
     return specPattern
   }
 
-  let dirname = parsedGlob.path.dirname
+  // Remove double-slashes from dirname (like if specPattern has /**/*/)
+  let dirname = parsedGlob.path.dirname.replaceAll(/\/\/+/g, '/')
 
-  if (dirname.startsWith('**')) {
-    dirname = dirname.replace('**', 'cypress')
-  }
+  // If a spec can be in any root dir, go ahead and use "cypress/"
+  if (dirname.startsWith('**')) dirname = dirname.replace('**', 'cypress')
 
   const splittedDirname = dirname.split('/').filter((s) => s !== '**').map((x) => replaceWildCard(x, testingType)).join('/')
   const fileName = replaceWildCard(parsedGlob.path.filename, 'filename')
 
   const extnameWithoutExt = parsedGlob.path.extname.replace(parsedGlob.path.ext, '')
+    || `.cy.${fileExtensionToUse}`
```

Comment:
I noticed a bug with a spec pattern like `cypress/component-tests/**/*` (no extension), this seemed to be the fix.

Reply:
> I noticed a bug with a spec pattern like cypress/component-tests/**/* (no extension), this seemed to be the fix.

What was the bug exactly?

Reply:
Check out this test:

https://github.com/cypress-io/cypress/blob/34fa8cbf5f9308bd7438df24b5c4bb0da423e60c/packages/data-context/test/unit/sources/ProjectDataSource.spec.ts#L437-L441

Before this change and the other change to remove double-slashes, the result here would be `cypress/component-tests//filename`. I assumed we don't want to give people a default filename with no extension or double slashes.